### ISSUE TITLE
workflow deletion UI loop and freeing disk troubles

### ISSUE THREAD

**Issue by tiborsimko** [reanahub/reana-ui#259]
**Current behaviour**

When a user is on the "Your workflows page" and would like to delete one workflow using the hamburger menu in the regular web interface, the user is brought to a new page that says:

![delete](https://user-images.githubusercontent.com/517546/162399125-af30472d-f7fe-4174-bef1-3e4ce43dbb1e.png)

Clicking on the "Delete" button does a job, sort of, because the workflow is deleted behind the scenes, but the page blinks white and the user is redirected to the very same page offering the Delete or Cancel options.

**Expected behaviour**

When the deletion is successful, the user would expect to see the confirmation and would expect to be brought back to the workflow list page from where the user started. 

**Note**

Whilst working on fixing this, it may be good to check up the "Free up disk" functionality from the hamburger menu on the detailed workflow page. It seems it is not working when trying to activate it after the above scenario.

![delete2](https://user-images.githubusercontent.com/517546/162400628-94ef3f56-60c3-4687-bc55-ee3c39c9ec1e.png)

**Comment by tiborsimko** [comment]
Update: WRT freeing up disk space observation, this appears to happen even when using CLI, as it were:

```console
$ reana-client list --verbose --include-workspace-size | grep af0ef30f-adbc-42a5-aa0b-bf31aa15c69b
root6-roofit-snakemake-kubernetes      1            2022-04-06T14:20:57   2022-04-06T14:21:10   2022-04-06T14:22:01   finished   af0ef30f-adbc-42a5-aa0b-bf31aa15c69b   00000000-0000-0000-0000-000000000000   1556480   2/2
$ find /var/reana/users/00000000-0000-0000-0000-000000000000/workflows | grep -c af0ef30f-adbc-42a5-aa0b-bf31aa15c69b
28

$ reana-client delete --include-workspace -w af0ef30f-adbc-42a5-aa0b-bf31aa15c69b
==> SUCCESS: af0ef30f-adbc-42a5-aa0b-bf31aa15c69b has been deleted

$ find /var/reana/users/00000000-0000-0000-0000-000000000000/workflows | grep -c af0ef30f-adbc-42a5-aa0b-bf31aa15c69b
0

$ reana-client list --verbose --include-workspace-size | grep af0ef30f-adbc-42a5-aa0b-bf31aa15c69b
root6-roofit-snakemake-kubernetes      1            2022-04-06T14:20:57   2022-04-06T14:21:10   2022-04-06T14:22:01   deleted    af0ef30f-adbc-42a5-aa0b-bf31aa15c69b   00000000-0000-0000-0000-000000000000   1556480   2/2
```

IOW, the workspace of the given workflow is well deleted, but the accounting shows it still occupies 1556480 bytes, which is not true.

It appears that the disk usage numbers are not updated "live". This may be due to reusing workflow termination policy configuration that are now set to "overnight" by default... However I think that in this case we need to act instantly, because  we are not terminating workflow pods, we are liberating work space via an explicit user command, usually done in order to launch new workflows asap.

(Leaving the observation here even though it'll obviously touch other non-UI components.)

**Comment by mdonadoni** [comment]
The issue regarding the deletion of a workflow was fixed by #257, which was merged yesterday. Can you reproduce this issue even with the latest `master` version of reana-ui?

**Comment by tiborsimko** [comment]
Indeed, I confirm that the "deletion loop" is fixed in `master` now :+1: The "free up disk" problem expressed in "Note" remains though; the workspace appears not to be deleted even though it is... Albeit that is an issue more for the backend, not for the frontend.  So the bulk of the work should go there.

Nevertheless, what we could improve WRT deletion on the web frontend side is the following scenario:
 
1. open some workflow details page
2. delete the workflow using hamburger menu
3. the page says "success!" but the otherwise stays the same, on the "running" state, and displayed in green. This may confuse users whether the deletion was really successful.
4. manually reload the page and see it getting "deleted" state and in grey

We could induce a refresh between step 3 and 4, so that no manual page reload would be necessary.  This will make it obvious that the workflow was indeed deleted. (BTW the page refresh could even be done automatically after 15 seconds via the polling, as @audrium recently introduced for created/queued/pending/running workflows in the details page.)
